### PR TITLE
feat: [FFM-3215] FF Java Server SDK 1.1.5.

### PR DESCRIPTION
**Pull request by milos85vasic**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/32629)

### CODE REVIEW

File: 960-continuous-features/src/main/java/io/harness/cf/CfClientModule.java [hunk 1]
```diff
@@ -46,39 +49,30 @@ CfClient provideCfClient(CfClientConfig cfClientConfig) {
...
+                                      .build();
+
+    final HarnessConnector harnessConnector = new HarnessConnector(cfClientConfig.getApiKey(), harnessConfig);
+
+    final BaseConfig config = BaseConfig.builder().bufferSize(5).analyticsEnabled(cfClientConfig.isAnalyticsEnabled()).build();
+
+    final CfClient client = new CfClient(harnessConnector, config);
+    try {
+      client.waitForInitialization();
+      final Target target =
```
Comment (davejohnston) [inline 960-continuous-features/src/main/java/io/harness/cf/CfClientModule.java:66]: I think this was possible debug/test code that should be removed?

File: 960-continuous-features/src/main/java/io/harness/cf/CfClientModule.java [hunk 2]
```diff
@@ -46,39 +49,30 @@ CfClient provideCfClient(CfClientConfig cfClientConfig) {
...
+              Target.builder()
+                      .identifier("target1")
+                      .isPrivate(false)
+                      .attribute("testKey", "TestValue")
+                      .name("target1")
+                      .build();
+    } catch (Exception e) {
+
```
Comment (davejohnston) [inline 960-continuous-features/src/main/java/io/harness/cf/CfClientModule.java:74]: Should we add a log in here to make it clear something went wrong

**Review by davejohnston — APPROVED** [review]
LGTM

File: 960-continuous-features/src/main/java/io/harness/cf/CfClientModule.java [hunk 3]
```diff
@@ -46,39 +48,23 @@ CfClient provideCfClient(CfClientConfig cfClientConfig) {
...
+                                      .eventUrl(cfClientConfig.getEventUrl())
+                                      .connectionTimeout(cfClientConfig.getConnectionTimeout())
+                                      .readTimeout(cfClientConfig.getReadTimeout())
+                                      .build();
 
-    final Supplier<Boolean> retrySupplier = Retry.decorateSupplier(
+    final HarnessConnector harnessConnector = new HarnessConnector(cfClientConfig.getApiKey(), harnessConfig);
 
-        retry, client::isInitialized);
+    final BaseConfig config = BaseConfig.builder().bufferSize(5).analyticsEnabled(cfClientConfig.isAnalyticsEnabled()).build();
```
Comment (enver-bisevac) [inline 960-continuous-features/src/main/java/io/harness/cf/CfClientModule.java:60]: Please remove this buffersize of 5 items

**Review by davejohnston — DISMISSED** [review]
> Reviewed 8 of 9 files at r4, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32629)* status: :shipit: complete! ✓

**Review by davejohnston — APPROVED** [review]
> Reviewed 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32629)* status: :shipit: complete! ✓

**Pull request merged**